webapp/channels/src/components/size_aware_image.tsx

```diff
@@ -23,83 +25,103 @@ const MIN_IMAGE_SIZE = 48;
 const MIN_IMAGE_SIZE_FOR_INTERNAL_BUTTONS = 100;
 const MAX_IMAGE_HEIGHT = 350;
 
+export type Props = {
+
+    /*
+    * The source URL of the image
+    */
+    src: string;
+
+    /*
+    * dimensions object to create empty space required to prevent scroll pop
+    */
+    dimensions?: Partial<PostImage>;
+    fileInfo?: FileInfo;
+
+    /**
+    * fileURL of the original image
+    */
+    fileURL?: string;
+
+    alt?: string;
+    height?: string;
+    width?: string;
+    title?: string;
+
+    /*
+    * Boolean value to pass for showing a loader when image is being loaded
+    */
+    showLoader?: boolean;
+
+    /*
+    * A callback that is called as soon as the image component has a height value
+    */
+    onImageLoaded?: ({height, width}: {height: number; width: number}) => void;
+
+    /*
+    * A callback that is called when image load fails
+    */
+    onImageLoadFail?: () => void;
+
+    /*
+    * Fetch the onClick function
+    */
+    onClick?: (e: (React.KeyboardEvent<HTMLImageElement> | React.MouseEvent<HTMLElement, MouseEvent>), link: string) => void;
```

Comment:
```suggestion
    onClick?: (e: (React.KeyboardEvent<HTMLImageElement> | React.MouseEvent<HTMLImageElement>), link: string) => void;
```

Reply:
some components are using the same click function on div elements So i used Element type

Reply:
then you can add either HtmlImageElement pr HtmlDivElement